### DISCUSSION TITLE
Need assistance with a transformation logic - Looks fine - somehow doesn't work :(

### DISCUSSION THREAD

**Discussion by aashang** [nextgenhealthcare/connect#4621] [Q&A] [answered]
This is for a ORM message,

for each (obr in msg.OBR)
{
var location= msg['PV1']['PV1.3']['PV1.3.1'].toString()
var locationproc = new Array("6HAD", "7HAD", "8HAD");

var obr7=parseInt(msg['MSH']['MSH.7']['MSH.7.1'].toString().substring(8,12),10)
var obr27=parseInt(msg['OBR']['OBR.27']['OBR.27.4'].toString().substring(8,12),10)

channelMap.put("location",location);
channelMap.put("locationproc",locationproc[0]);
channelMap.put("obr7",obr7);
channelMap.put("obr27",obr27);
var log="A"
if (msg['MSH']['MSH.9']['MSH.9.1'].toString()=='ORM')
{
if (location=='6HAD' || location=='7HAD'||location=='8HAD') 
{
log="Step1"
channelMap.put("log",log)

if (((obr['OBR.27']['OBR.27.6']=='R')||(obr['OBR.27']['OBR.27.6']=='T'))) 
{	
	log="Step2A"
	channelMap.put("log",log)
	
if ((obr27>=0400) && (obr27<=0700))
{
log="Step2B"
channelMap.put("log",log)	
	if ((obr7<=0400) || (obr7>=0700))
	{
	obr['OBR.11']['OBR.11.1']='L' //  Labcollect
	log="Step3"
	channelMap.put("log",log)
	}

}
}
}
}
channelMap.put("log",log)
}

So till Step2A it works fine, however even for a qualifying condition it fails to go to step 2B.

I am so confused why :(

![image](https://user-images.githubusercontent.com/86919722/124388758-4fde5080-dcf5-11eb-95ae-558124191b4c.png)

**Comment by pacmano1** [comment, marked as answer]
That first obr27 assignment is not using your iteration var from the for loop.  Is that intentional?

**Reply by aashang** [reply]
@pacmano1  Yes, it is just so that it loops if in case there is more than one OBR segment per ORM message.

**Reply by pacmano1** [reply]
Post a sample message and your objective for the parsing in plain terms.

**Reply by aashang** [reply]
@pacmano1 , we got it to work, the fix was odd, we just removed the leading 0's in the numerical values we are using to compare in our if logic.

eg:
instead of using 

if ((obr27>=0400) && (obr27<=0700))

I used

if ((obr27 >= 400) && (obr27 <= 700))

and that resolved the issue. Not sure the root cause here.

Sorry I am very new to Javascripts, so I might sound like a noob :)
